File: Master_the_Key_ROS_Concepts/Lec1_Create_a_Cakin_Workspace/catkin_ws/src/my_robot_tutorials/scripts/robot_news_radio_transmitter.py
#!/usr/bin/env python3

import rospy
from std_msgs.msg import String #simply import string type for topic

if __name__ == '__main__':
    #first initialize node
    rospy.init_node('robot_news_radio_transmitter')

    #we want to publish topic
    pub = rospy.Publisher("/robot_news_radio", String, queue_size=10) #inside () we want to name the topic
    # rospy.Publisher(name of topic, message type, size)
    # now publisher is ready


    rate = rospy.Rate(2) #publish every 2Hz

    # publish the data now!
    while not rospy.is_shutdown():
        msg = String()
        msg.data = "Hi, this is Howard from the Robot News Radio!"#msg.data because .data is where String in std_msgs.msg library
        
        #save publisher
        pub.publish(msg)

        rate.sleep()

    rospy.loginfo("Node was topped")

File: Master_the_Key_ROS_Concepts/Lec1_Create_a_Cakin_Workspace/catkin_ws/src/my_robot_tutorials/src/robot_news_radio_transmitter.cpp
#include <ros/ros.h>
#include <std_msgs/String.h> //include header of the msg file

int main(int argc, char **argv)
{
	//Create node
	ros::init(argc, argv, "robot_news_radio_transmitter");
	ros::NodeHandle nh;
	
	ROS_INFO("Node has been started");
	
	// Begin publisher object
	ros::Publisher pub = nh.advertise<std_msgs::String>("/robot_news_radio", 10); //Publisher object = nodehandle.advertise<type of message>(name of the topic, queue size)

	ros::Rate rate(3);

	while (ros::ok()){
		std_msgs::String msg;//create new msg
		msg.data = "Hi, This is Howard from the Robot News Radio";
		pub.publish(msg);
		rate.sleep();
	}
}
